Source/WebCore/platform/graphics/cg/ImageUtilitiesCG.cpp

```diff
@@ -164,4 +166,95 @@ Expected<std::pair<String, Vector<IntSize>>, ImageDecodingError> utiAndAvailable
     return std::make_pair(WTFMove(uti), WTFMove(sizes));
 }
 
+Vector<uint8_t> createIconFromImageData(std::span<const uint8_t> data, std::span<const unsigned> lengths)
+{
+    Ref buffer = FragmentedSharedBuffer::create(data);
+    Ref imageDecoder = ImageDecoderCG::create(buffer.get(), AlphaOption::Premultiplied, GammaAndColorProfileOption::Applied);
+    imageDecoder->setData(buffer.get(), true);
+    if (imageDecoder->encodedDataStatus() == EncodedDataStatus::Error)
+        return { };
+
+    auto sourceUTI = imageDecoder->uti();
+    if (!isSupportedImageType(sourceUTI))
+        return { };
+
+    // If the image contains multiple frames, we don't know which one to use.
+    size_t frameCount = imageDecoder->frameCount();
+    if (frameCount != 1)
+        return { };
+
+    RetainPtr image = imageDecoder->createFrameImageAtIndex(0);
```

Comment:
Should not we get the frame of the primary index instead of the first frame?

Reply:
I think primary index equals to first index since the image should only contains one frame; but I think using primary index and removing the requirement of "one frame" would also work!